draft-aegis-mimi-arch-latest.md

```diff
@@ -560,17 +564,38 @@ Output:
 
 The above request has the following effect:
 
-* An arbitrary key pool with attributes matching the provided `appId`,
+// TODO: Should we really have isFinalKey or is that potentially a security
+problem and we should say that the pool should just return an error in the event
+there are no more keys.
+
+* A key queue with attributes matching the provided `appId`,
   `clientId`, `protocolVersion`, `cipherSuite` and `appLabel` is chosen. If
-  there is no such key pool, the request has no effect.
-* If the above key pool contains (non-final) key packages, a key package from
-  the pool is chosen (according to gatewey's policy), returned and deleted. The
-  `isFinalKey` flag in the output is set to false. The returned key package MUST
-  NOT be returned by any future fetch key package request.
-* Else, if there is a final key stored for the key pool, the output contains
+  there is no such key queue, the request has no effect.
+* If a non-final key in the queue is assigned to `receiver_client_id`, and that
+  key package is not expired then that key should be returned.
```

Comment:
Doesn't the above "If a non-final..." text mean that Alice only gets one KP for Bob until his queue is advanced? When is advancing done? What if Alice wants to invite Bob to 2 groups while he's offline?

Reply:
I left a note that it is advanced based on the rules in the now non-existent group operation section about processing proposals / commits. You can invite to 2 groups because as soon as you send a proposal / commit that uses one key, you can go grab another new key